MapleServer2/Types/Mailbox.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+
+namespace MapleServer2.Types
+{
+    public class Mailbox
+    {
+        public List<Mail> Box { get; private set; }
+
+        public Mailbox()
+        {
+            Box = new List<Mail>();
+        }
+
+        public void AddOrUpdate(Mail mail)
+        {
+            int index = Box.FindIndex(x => x.Uid == mail.Uid);
+            if (index > -1)
+            {
+                Box[index] = mail;
+            }
+            else
+            {
+                Box.Insert(0, mail); // Adds to front of mailbox
+            }
+        }
+
+        public long Read(int id)
+        {
+            int index = Box.FindIndex(x => x.Uid == id);
+            Box[index].Read(DateTimeOffset.UtcNow.ToUnixTimeSeconds());
+
+            return Box[index].ReadTimestamp;
+        }
+
+        public List<Item> Collect(int id)
+        {
+            int index = Box.FindIndex(x => x.Uid == id);
+            List<Item> items = Box[index].Items;
+            Box[index].Collect(null);
+
+            return items;
+        }
+
+        public int GetUnread()
```

Comment:
I feel like the method names here should better describe what the method is actually doing, for example `Read` might be better as `GetReadTimestamp`, `Collect` as `GetMailById` and `GetUnread` as `GetUnreadCount` since that can easily be interpreted as getting all the unread mail, and not simply the count of them.

Reply:
`Read` is setting the `ReadTimestamp` though, and `Collect` sets the `Items`. `GetUnread` could definitely be renamed to `GetUnreadCount` though.